### PR TITLE
Add support for custom schemes

### PR DESCRIPTION
**Pull request by rowanmanning**
See #11 for more information.

### CODE REVIEW

File: lib/image-transform.js [hunk 1]
```diff
@@ -156,6 +158,54 @@ module.exports = class ImageTransform {
...
+		baseUrl = baseUrl.replace(/\/+$/, '');
+		pathname = pathname.replace(/\/+$/, '');
+
+		let version;
+		if (/\-v\d+$/.test(scheme)) {
+			scheme = scheme.split('-');
+			version = scheme.pop();
+			scheme = scheme.join('-');
+		}
+		version = version || 'unversioned';
```
Comment (AlbertoElias) [inline lib/image-transform.js]: I find hard to follow all the transformations and info extraction that's going on here, could it possible have some more comments?

File: lib/routes/v2/images-debug.js [hunk 2]
```diff
@@ -1,15 +1,34 @@
...
+				transform: request.transform,
+				appliedTransform: request.appliedTransform
+			});
+		}
+	);
+
+	// Image with a custom scheme
+	// TODO make sure these default to the original exension! E.g. SVG for fticon
+	app.get(
+		/^\/v2\/images\/debug\/((ftcms|fticon|fthead|ftsocial|ftpodcast|ftlogo)(\-v\d+)?(:|%3A).*)$/,
```
Comment (AlbertoElias) [inline lib/routes/v2/images-debug.js]: same for regexps, there are quite a few

File: lib/image-transform.js (hunk 1 above)
Comment (rowanmanning) [inline lib/image-transform.js]: Yeah absolutely. Sorry, I've got out of the habit which is bad!

File: test/unit/lib/image-transform.js [hunk 3]
```diff
@@ -497,6 +505,213 @@ describe('lib/image-transform', () => {
...
+
+			it('throws an error', () => {
+				assert.throws(() => {
+					ImageTransform.resolveCustomSchemeUri('foo:example', 'http://base/images');
+				}, 'Image URI must be a string with a valid scheme');
+			});
+
+		});
+
+		describe('when `uri` doesn\'t have a hostname', () => {
```
Comment (AlbertoElias) [inline test/unit/lib/image-transform.js]: The error is the same as above, and why is icon name called hostname?

Comment (rowanmanning) [inline test/unit/lib/image-transform.js]: Yeah I'm making sure that the error gets triggered properly on different cases. It's called a hostname in the test because when you parse it as a URL the icon name comes through as `parsedUrl.hostname`. I can change if it's confusing

File: lib/image-transform.js [hunk 4]
```diff
@@ -158,45 +266,71 @@ module.exports = class ImageTransform {
...
 			case 'fticon':
 			case 'ftlogo':
 			case 'ftsocial':
 				if(!path.extname(pathname)) {
 					pathname = `${pathname}.svg`;
 				}
 				return `${baseUrl}/${scheme}/${version}/${pathname}`;
+
+			// PNG-based schemes. Here we're mostly adding
+			// the .svg extension if there isn't an extension
```
Comment (AlbertoElias) [inline lib/image-transform.js]: .png*

**Comment by AlbertoElias** [comment]
left one comment, rest looks good!
